packages/store/src/storeHookTypes.sol

```diff
@@ -3,8 +3,9 @@ pragma solidity >=0.8.0;
 
 uint8 constant BEFORE_SET_RECORD = 1 << 0;
 uint8 constant AFTER_SET_RECORD = 1 << 1;
-// TODO: do we need to differentiate between static and dynamic set field?
-uint8 constant BEFORE_SET_FIELD = 1 << 2;
-uint8 constant AFTER_SET_FIELD = 1 << 3;
-uint8 constant BEFORE_DELETE_RECORD = 1 << 4;
-uint8 constant AFTER_DELETE_RECORD = 1 << 5;
+uint8 constant BEFORE_SPLICE_STATIC_DATA = 1 << 2;
+uint8 constant AFTER_SPLICE_STATIC_DATA = 1 << 3;
+uint8 constant BEFORE_SPLICE_DYNAMIC_DATA = 1 << 4;
+uint8 constant AFTER_SPLICE_DYNAMIC_DATA = 1 << 5;
+uint8 constant BEFORE_DELETE_RECORD = 1 << 6;
+uint8 constant AFTER_DELETE_RECORD = 1 << 7;
```

Comment:
this is what I was referring to btw when I thought we maybe needed more than 8 slots for hooks

do we wanna consider a bump to uint16 or uint32 for future proofing?

Reply:
I'd prefer to only bump to `uint16` once we actually need it (don't think we'll need it for this iteration of the protocol) - although one could argue we could avoid a breaking change if we already use `uint16` now and later add additional functionality for other types of hook 🤔

Reply:
yeah my main motivation is reducing breaking changes later

I guess either way it'd require a redeploy of the world, so maybe it's fine to leave this and expand when we need it